### PR TITLE
Add 'PreserveSpaces' opt

### PR DESCRIPTION
**Pull request by AndrewPecha**
This is driven by [this issue from a few days ago](https://github.com/moov-io/ach/issues/1159)

- Added `SetValidation` to `BatchControl` so opts could be set before parsing
- Added `parseStringFieldWithOpts` to `converters` struct
- Modified parsing for `FileHeader`, `BatchHeader`, and `BatchControl`  to respect the new opt where spaces were normally trimmed
- Modified the parsing logic for `EntryDetail` to not trim spaces for RDFI, CheckDigit, and Trace Number (I believe these all need numeric values and use every character to be considered valid anyway, but please correct me if I'm wrong)

### CODE REVIEW

File: batchControl_test.go [hunk 1]
```diff
@@ -315,3 +315,24 @@ func BenchmarkBatchControlLength(b *testing.B) {
...
+func TestBatchControl__SetValidation(t *testing.T) {
+	bh := NewBatchControl()
+	bh.SetValidation(&ValidateOpts{})
+
+	// nil out
+	bh = nil
+	bh.SetValidation(&ValidateOpts{})
+}
+
+func TestBatchControl__IsValidWithPreserveSpacesOpt(t *testing.T) {
```
Comment (AndrewPecha) [inline batchControl_test.go:328]: Wanted to ensure nothing relied on spaces being trimmed when validating

File: entryDetail.go [hunk 2]
```diff
@@ -204,9 +204,9 @@ func (ed *EntryDetail) Parse(record string) {
 	// 2-3 is checking credit 22 debit 27 savings credit 32 debit 37
 	ed.TransactionCode = ed.parseNumField(record[1:3])
 	// 4-11 the RDFI's routing number without the last digit.
-	ed.RDFIIdentification = ed.parseStringField(record[3:11])
+	ed.RDFIIdentification = record[3:11]
```
Comment (AndrewPecha) [inline entryDetail.go:207]: I think we are safe to drop the space trimming here altogether, but please correct me if I'm wrong

Comment (adamdecaf) [reply entryDetail.go:207]: Yea. It's probably fine with some fields.

**Review by adamdecaf — APPROVED** [review]

File: batchControl_test.go (hunk 1 above)
Comment (adamdecaf) [reply batchControl_test.go:328]: Thanks

**Pull request merged**